01_stratified_classification/nbconverted/lasso_range_all_optimizers.py

```diff
@@ -0,0 +1,393 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### LASSO parameter range experiments, summary across all genes
+
+# In[1]:
+
+
+import os
+import itertools as it
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib.colors import Normalize
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+ll_results_dir = os.path.join(
+    cfg.repo_root, '01_stratified_classification', 'results', 'optimizer_compare_ll', 'gene'
+)
+
+sgd_results_dir = os.path.join(
+    cfg.repo_root, '01_stratified_classification', 'results', 'optimizer_compare_sgd', 'gene'
+)
+
+plot_gene = 'EGFR'
+metric = 'aupr'
+
+output_plots = True
+output_plots_dir = os.path.join(
+    cfg.repo_root, '01_stratified_classification', 'optimizers_plots'
+)
+
+
+# ### Get coefficient information for each lasso penalty
+
+# In[3]:
+
+
+ll_nz_coefs_df = []
+
+# get coefficient info for training dataset specified above
+for coef_info in au.generate_nonzero_coefficients_lasso_range(ll_results_dir):
+    (gene,
+     cancer_type,
+     seed,
+     lasso_param,
+     coefs_list) = coef_info
+    for fold_no, coefs in enumerate(coefs_list):
+        ll_nz_coefs_df.append(
+            [gene, cancer_type, lasso_param, seed, fold_no, len(coefs)]
+        )
+        
+ll_nz_coefs_df = pd.DataFrame(
+    ll_nz_coefs_df,
+    columns=['gene', 'signal', 'lasso_param', 'seed', 'fold', 'nz_coefs']
+)
+ll_nz_coefs_df.lasso_param = ll_nz_coefs_df.lasso_param.astype(float)
+
+print(ll_nz_coefs_df.gene.unique())
+ll_nz_coefs_df.head()
+
+
+# In[4]:
+
+
+sgd_nz_coefs_df = []
+
+# get coefficient info for training dataset specified above
+for coef_info in au.generate_nonzero_coefficients_lasso_range(sgd_results_dir):
+    (gene,
+     cancer_type,
+     seed,
+     lasso_param,
+     coefs_list) = coef_info
+    for fold_no, coefs in enumerate(coefs_list):
+        sgd_nz_coefs_df.append(
+            [gene, cancer_type, lasso_param, seed, fold_no, len(coefs)]
+        )
+        
+sgd_nz_coefs_df = pd.DataFrame(
+    sgd_nz_coefs_df,
+    columns=['gene', 'signal', 'lasso_param', 'seed', 'fold', 'nz_coefs']
+)
+sgd_nz_coefs_df.lasso_param = sgd_nz_coefs_df.lasso_param.astype(float)
+
+print(sgd_nz_coefs_df.gene.unique())
+sgd_nz_coefs_df.head()
+
+
+# ### Get performance information for each lasso penalty
+
+# In[5]:
+
+
+ll_perf_df = au.load_prediction_results_lasso_range(ll_results_dir,
+                                                    'stratified')
+ll_perf_df = (ll_perf_df
+    .drop(columns=['holdout_cancer_type', 'experiment'])
+    .copy()
+)
+ll_perf_df.lasso_param = ll_perf_df.lasso_param.astype(float)
+
+print(ll_perf_df.shape)
+print(ll_perf_df.gene.unique())
```

Comment:
Do you need to print this in every code block?

Reply:
I generally do this to catch bugs with how the data is being read or generated - I've found that it helps to eyeball what the dataset looks like in terms of shape and columns included, etc. I guess in this case it's most important that I generated data with all of the same genes for SGD and liblinear, since I wouldn't expect them to have exactly the same number of rows due to differences, so maybe I can explicitly check that and make a note of it.